gadgets/fsnotify/test/unit/fsnotify_test.go

```diff
@@ -0,0 +1,185 @@
+		"captures_inotify_event": {
+			runnerConfig:  runnerConfig,
+			generateEvent: generateEvent,
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, filename string, events []ExpectedFsnotifyEvent) {
+				utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, pid int) *ExpectedFsnotifyEvent {
+					return &ExpectedFsnotifyEvent{
+						Type: "inotify",
+
+						IMask: 134217732, // 134217732 = 0x08000004 = FS_ATTRIB | FS_EVENT_ON_CHILD
+						Name:  filename,
+
+						Timestamp: utils.NormalizedStr,
+
+						TraceeMntnsId: utils.NormalizedInt,
+						TracerMntnsId: utils.NormalizedInt,
+
+						FaType:     utils.NormalizedStr,
+						FaResponse: utils.NormalizedStr,
+
+						IWd:     utils.NormalizedInt,
+						IIno:    utils.NormalizedInt,
+						IInoDir: utils.NormalizedInt,
+					}
+				})(t, info, 0, events)
+			},
+		},
+	}
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var filename string
+			runner := utilstest.NewRunnerWithTest(t, testCase.runnerConfig)
+
+			normalizeEvent := func(event *ExpectedFsnotifyEvent) {
+				utils.NormalizeString(&event.Timestamp)
+
+				utils.NormalizeProc(&event.TraceeProc)
+				utils.NormalizeProc(&event.TracerProc)
+				utils.NormalizeInt(&event.TraceeMntnsId)
+				utils.NormalizeInt(&event.TracerMntnsId)
+
+				utils.NormalizeInt(&event.TraceeUId)
+				utils.NormalizeInt(&event.TraceeGId)
+				utils.NormalizeInt(&event.TracerUId)
+				utils.NormalizeInt(&event.TracerGId)
+
+				utils.NormalizeInt(&event.Prio)
+				utils.NormalizeInt(&event.FaMask)
+
+				utils.NormalizeString(&event.FaType)
+				utils.NormalizeInt(&event.FaPId)
+				utils.NormalizeInt(&event.FaFlags)
+				utils.NormalizeInt(&event.FaFFlags)
+				utils.NormalizeString(&event.FaResponse)
+
+				utils.NormalizeInt(&event.IWd)
+				utils.NormalizeInt(&event.ICookie)
+				utils.NormalizeInt(&event.IIno)
+				utils.NormalizeInt(&event.IInoDir)
+			}
+			onGadgetRun := func(gadgetCtx operators.GadgetContext) error {
+				utilstest.RunWithRunner(t, runner, func() error {
+					var err error
+					filename, err = testCase.generateEvent()
+					if err != nil {
+						return err
+					}
+					return nil
+				})
+				return nil
+			}
+			opts := gadgetrunner.GadgetRunnerOpts[ExpectedFsnotifyEvent]{
+				Image:          "fsnotify",
+				Timeout:        5 * time.Second,
+				OnGadgetRun:    onGadgetRun,
+				NormalizeEvent: normalizeEvent,
+			}
+
+			gadgetRunner := gadgetrunner.NewGadgetRunner(t, opts)
+
+			gadgetRunner.RunGadget()
+
+			testCase.validateEvent(t, runner.Info, filename, gadgetRunner.CapturedEvents)
+		})
+	}
+}
+
+func generateEvent() (string, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return "", err
+	}
+	defer watcher.Close()
+
+	err = watcher.Add("/tmp/")
+	if err != nil {
+		return "", err
+	}
+
+	touchCmd := exec.Command("touch", "/tmp/ABCDE")
+	err = touchCmd.Run()
+	if err != nil {
+		return "", err
+	}
```

Comment:
What about doing it directly from this executable instead of relying on an external binary?

Also, what about using a true temporal file (by using [os.CreateTemp](https://pkg.go.dev/os#CreateTemp)) to avoid any possible collision.

Reply:
Yes, that works well
I’ve updated the `generateEvent` method to use `os.CreateTemp`

Thanks :)